buildspec.pypi.yml

```diff
@@ -0,0 +1,36 @@
+version: 0.2
+
+env:
+  shell: bash
+  parameter-store:
+    PYPI_TOKEN: "pypi-token"
+
+phases:
+  install:
+    commands:
+      - echo Setting local Python versions
+      - pyenv versions | awk 'match($0, /\d\.\d+\.\d+/) { print substr($0, RSTART, RLENGTH) }' | sort -rV > .python-version
+      - echo Installing dependencies
+      - pip install poetry
+      - poetry install
+
+  build:
+    commands:
+      - echo Running check-cloudformation
```

Comment:
Do we want this to only happen against merges to `main`?

Reply:
We have another CodeBuild build project that does this on PRs: https://github.com/uktrade/copilot-tools/blob/main/buildspec.test.yml

Reply:
`buildspec.test.yml` for PR branches and `buildspec.pypi.yml` on `main`.

Reply:
Sorry, not the CFN linting, I mean do we want it to public to pypi only on merging into main?  Or is this how the codebuild project will bet set up?

Reply:
Yes, I think so. If we had multiple PRs open, then the version bumps may conflict with each other, so merging into `main` and then triggering this pipeline would be better, IMO, as the other PR branches would have to merge/rebase `main`.